packages/btc/src/api/sendRbf.ts

```diff
@@ -0,0 +1,169 @@
+import { BaseOutput, Utxo } from '../transaction/utxo';
+import { DataSource } from '../query/source';
+import { ErrorCodes, TxBuildError } from '../error';
+import { InitOutput, TxBuilder } from '../transaction/build';
+import { isOpReturnScriptPubkey } from '../transaction/embed';
+import { networkTypeToNetwork } from '../preset/network';
+import { networkTypeToConfig } from '../preset/config';
+import { createSendUtxosBuilder } from './sendUtxos';
+import { isP2trScript } from '../script';
+import { bitcoin } from '../bitcoin';
+
+export interface SendRbfProps {
+  from: string;
+  txHex: string;
+  source: DataSource;
+  feeRate?: number;
+  fromPubkey?: string;
+  changeIndex?: number;
+  changeAddress?: string;
+  minUtxoSatoshi?: number;
+  onlyConfirmedUtxos?: boolean;
+  requireValidOutputsValue?: boolean;
+  requireGreaterFeeAndRate?: boolean;
+
+  // EXPERIMENTAL: the below props are unstable and can be altered at any time
+  inputsPubkey?: Record<string, string>; // Record<address, pubkey>
+}
+
+export async function createSendRbfBuilder(props: SendRbfProps): Promise<{
+  builder: TxBuilder;
+  fee: number;
+  feeRate: number;
+  changeIndex: number;
+}> {
+  const previousTx = bitcoin.Transaction.fromHex(props.txHex);
+  const network = networkTypeToNetwork(props.source.networkType);
+
+  // Rebuild inputs
+  const inputs: Utxo[] = [];
+  for (const input of previousTx.ins) {
+    const hash = Buffer.from(input.hash).reverse().toString('hex');
+    const utxo = await props.source.getUtxo(hash, input.index);
+    if (!utxo) {
+      throw TxBuildError.withComment(ErrorCodes.CANNOT_FIND_UTXO, `hash: ${hash}, index: ${input.index}`);
+    }
+
+    // Ensure each P2TR input has a corresponding pubkey
+    const fromPubkey = utxo.address === props.from ? props.fromPubkey : undefined;
+    const inputPubkey = props.inputsPubkey?.[utxo.address];
+    const pubkey = inputPubkey ?? fromPubkey;
+    if (pubkey) {
+      utxo.pubkey = pubkey;
+    }
+    if (isP2trScript(utxo.scriptPk) && !utxo.pubkey) {
+      throw TxBuildError.withComment(ErrorCodes.MISSING_PUBKEY, utxo.address);
+    }
+
+    inputs.push(utxo);
+  }
+
+  // Rebuild outputs
+  const requireValidOutputsValue = props.requireValidOutputsValue ?? false;
+  const outputs: InitOutput[] = previousTx.outs.map((output) => {
+    if (isOpReturnScriptPubkey(output.script)) {
+      return {
+        script: output.script,
+        value: output.value,
+        fixed: true,
+      };
+    } else {
+      return {
+        minUtxoSatoshi: requireValidOutputsValue ? undefined : output.value,
+        address: bitcoin.address.fromOutputScript(output.script, network),
+        value: output.value,
+        fixed: true,
+      };
```

Comment:
Since this output is `fixed`, what is the purpose of `minUtxoSatoshi` and `requireValidOutputsValue`?

Reply:
The `fixed` prop specifies that the output.value is undeductible, while the `minUtxoSatoshi` prop specifies the minimal satoshi of the output. By default, if the original transaction has an output with 546 satoshi of value, an error will be thrown since the default value of minUtxoSatoshi is `1000` on the testnet and `output.value < minUtxoSatoshi`. 

To avoid this error, the `sendRbf()` API disables the validation by default, and adds a `requireValidOutputsValue` prop to allow users to re-enable the validation manually.

Reply:
I see.